pkg/doc/bbs/bbs12381g2pub/bbs.go

```diff
@@ -83,7 +88,61 @@ func (b BBSG2Pub) Verify(messages [][]byte, sigBytes, pubKeyBytes []byte) error
 	return errors.New("BLS12-381: invalid signature")
 }
 
-func getB(s *bls.FR, messages []*SignatureMessage, key *PublicKey) (*bls.G1Affine, error) {
+// Sign signs the one or more messages using private key in compressed form.
+func (bbs *BBSG2Pub) Sign(messages [][]byte, privKeyBytes []byte) ([]byte, error) {
+	privKey, err := UnmarshalPrivateKey(privKeyBytes)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal private key: %w", err)
+	}
+
+	if len(messages) == 0 {
+		return nil, errors.New("messages are not defined")
+	}
+
+	return bbs.SignWithKeyPair(messages, privKey, privKey.PublicKey())
+}
+
+// SignWithKeyPair signs the one or more messages using BBS+ key pair.
+func (bbs *BBSG2Pub) SignWithKeyPair(messages [][]byte, privKey *PrivateKey, pubKey *PublicKey) ([]byte, error) {
```

Comment:
doesn't `privKey` include the public key as well ? can't we just pass privKey alone then do privKey.PublicKey() inside the function to fetch its public key?

what happens if someone passes a public key not matching the private one? I think its unnecessary to pass pubKey to this function.

Reply:
I agree with this point. No need to pass a `pubKey` as it's created from `privKey` by a straightforward algorithm.

Reply:
Corrected the API.